crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1885,36 +1885,43 @@ pub async fn retrieve_payment_method(
 pub async fn delete_payment_method(
     state: &routes::AppState,
     merchant_account: domain::MerchantAccount,
-    pm: api::PaymentMethodId,
+    pm_id: api::PaymentMethodId,
 ) -> errors::RouterResponse<api::PaymentMethodDeleteResponse> {
-    let (_, supplementary_data) =
-        vault::Vault::get_payment_method_data_from_locker(state, &pm.payment_method_id).await?;
```

Comment:
Why is this removed? Please have a look if this is necessary and add it back

Reply:
Please read the description.   

To delete from Vault (locker), we need `payment_token` which can be obtained when doing `save card flow` or `mandates`. Other than that, we're not storing the `payment_method_id` or its data. When we create a `payment_method`, we create its related `id` and never store that in the locker but rather database. So it is useless checking there and it just throws 500.